### PR TITLE
Fixed user reporting pagination

### PR DESCRIPTION
**Pull request by harshilsharma63**
#### Summary
Fixed issue where no results were returned when sorting in descending order, and going to previous page didn't work.

#### Ticket Link
Fixes https://mattermost.atlassian.net/browse/MM-56281

#### Release Note
```release-note
none
```

### CODE REVIEW

File: server/channels/store/sqlstore/user_store.go [hunk 1]
```diff
@@ -2288,37 +2288,46 @@ func (us SqlUserStore) GetUserReport(filter *model.UserReportOptions) ([]*model.
...
+				sq.Gt{filter.SortColumn: filter.FromColumnValue},
+				sq.And{
+					sq.Eq{filter.SortColumn: filter.FromColumnValue},
+					sq.Gt{"u.Id": filter.FromId},
+				},
+			})
+		}
 	}
 
+	query = query.OrderBy(filter.SortColumn+" "+sortDirection, "u.Id")
```
Comment (larkox) [inline server/channels/store/sqlstore/user_store.go:2329]: Nit: using `fmt.Sprintf` may make this a bit more legible, but 0/5.

File: server/channels/store/sqlstore/user_store.go [hunk 2]
```diff
@@ -2369,8 +2378,26 @@ func (us SqlUserStore) GetUserReport(filter *model.UserReportOptions) ([]*model.
...
+		if sortDirection == "ASC" {
+			reverseSortDirection = "DESC"
+		}
+
+		parentQuery = us.getQueryBuilder().
+			Select("*").
+			FromSelect(query, "data").
+			OrderBy(filter.SortColumn+" "+reverseSortDirection, "Id")
+	}
+
```
Comment (larkox) [inline server/channels/store/sqlstore/user_store.go:2398]: Is it cheaper to reverse on database than reverse on memory?

**Review by larkox — APPROVED** [review]
LGTM, just two small considerations

File: server/channels/store/sqlstore/user_store.go [hunk 3]
```diff
@@ -2369,8 +2378,26 @@ func (us SqlUserStore) GetUserReport(filter *model.UserReportOptions) ([]*model.
 		query = query.Where(sq.Eq{"u.DeleteAt": 0})
 	}
 
+	parentQuery := query
+	// If we're going a page back...
+	//
+	// The way pagination works, we get the previous page's rows
+	// in reverse order. So, we use parent query on it to
+	// reverse the order in database itself.
+	if filter.Direction == "up" {
```
Comment (M-ZubairAhmed) [inline server/channels/store/sqlstore/user_store.go:2387]: we thought of renaming these to prev/next, can we do that in this PR or next

**Review by M-ZubairAhmed — CHANGES_REQUESTED** [review]

**Review by M-ZubairAhmed — APPROVED** [review]

**Pull request merged**